.github/workflows/main.yml

```
@@ -38,32 +39,60 @@
+          name: coverage-reports
+          path: CoverageReports
 
   publish:
     name: Publish Application
     runs-on: ubuntu-latest
     needs: build
+    timeout-minutes: 20  # ⏱️ Publishing timeout
 
     steps:
       - name: Checkout Code
         uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)